atc/steps.go

```diff
@@ -126,6 +132,8 @@ func (step Step) MarshalJSON() ([]byte, error) {
 }
 
 // See the note about json tags here: https://golang.org/pkg/encoding/json/#Marshal
+// keys are deleted based on the assumption that the first non-ignorable field
+// on the step struct matches the name of the step (e.g. get, put, on_success, etc.)
```

Comment:
Not sure I totally understand this line. This function deletes every field that the step's `struct` defines that is unmarshaled (i.e. isn't ignored using `json:"-"`). I guess the assumption is that the `Key` in `StepPrecedence` is one of the fields that is declared on the `struct`. Though saying "the first non-ignorable field" is a bit misleading I think

Reply:
Hmm I'll change this comment.

> function deletes every field that the step's struct defines 

This line confuses me and I can interpret it in two different ways, not sure which one you mean. I'll update the comment, hopefully it'll clear things up!

Reply:
Okay I think I cleared it up! I actually removed the comment and renamed the function instead. I see how my initial comment was misleading. Let me know your thoughts!

Reply:
The changes make sense to me!